File: Backend/src/main/java/com/github/francisfire/anavis/models/ActivePrenotation.java
package com.github.francisfire.anavis.models;

import java.util.Date;

import org.springframework.data.mongodb.core.mapping.Document;

import lombok.Setter;

import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@Setter
@NoArgsConstructor
@Document(collection = "activeprenotation")
public class ActivePrenotation extends Prenotation {

	private boolean confirmed;

	public ActivePrenotation(String id, String officeMail, String donorMail, Date hour, boolean confirmed) {
		super(id, officeMail, donorMail, hour);
		this.confirmed = confirmed;
	}

}

File: Frontend/lib/models/activeprenotation.dart
import 'package:anavis/models/prenotation.dart';

class ActivePrenotation extends Prenotation {
  bool _confirmed;
  ActivePrenotation(
      String id, String officeMail, String donorMail, String hour, bool conf)
      : _confirmed = conf,
        super.withParams(id, officeMail, donorMail, hour);
  bool isConfirmed() {
    return _confirmed;
  }
}
